packages/workflow-compile/new/index.js

```diff
@@ -60,31 +60,51 @@ async function compile(config) {
     {}
   );
 
+  const [compilerUsed] = Object.values(compilations)
+    .map(({ compilerInfo }) => compilerInfo)
+    .filter(compilerInfo => compilerInfo);
+
   const contracts = Object.values(compilations)
     .map(({ contracts }) => contracts)
     .reduce((a, b) => [...a, ...b], []);
 
-  return { contracts, compilations };
+  return { contracts, compilations, compilerUsed };
 }
 
 const Contracts = {
   async compile(options) {
     const config = prepareConfig(options);
-    reportCompilationStarted(config);
 
-    const { contracts, compilations } = await compile(config);
+    if (config.events) config.events.emit("compile:start");
+
+    const { contracts, compilations, compilerUsed } = await compile(config);
+
+    if (compilerUsed) {
+      config.compilersInfo[compilerUsed.name] = {
+        version: compilerUsed.version
+      };
+    }
 
-    if (contracts.length === 0) {
-      reportNothingToCompile(config);
+    if (contracts.length === 0 && config.events) {
+      config.events.emit("compile:nothingToCompile");
     }
 
-    reportCompilationFinished(config);
+    if (config.events) {
+      config.events.emit("compile:succeed", {
+        contractsBuildDirectory: config.contracts_build_directory,
+        compilersInfo: config.compilersInfo
+      });
+    }
     return {
       contracts,
       compilations
     };
   },
 
+  reportCompilationStarted,
+  reportCompilationFinished,
+  reportNothingToCompile,
```

Comment:
Why are these getting exported here now?

Reply:
Not sure, saw that they were exported [here](https://github.com/trufflesuite/truffle/blob/19bf2462e51ea609267b9a8c8bc951e9c2ce09b5/packages/workflow-compile/legacy/index.js#L113-L115).

Reply:
Haven't checked if we even use these methods anymore. Guess we could remove them in a separate PR.

Reply:
The core logic getting fixed in this PR should probably go out in this week's release.

Reply:
Agreed

Reply:
Oh, interesting, it looks like we export these functions in `workflow-compile/legacy` because the Truffle Library API makes them available (`truffle.contracts.reportCompilationsStarted`, etc..).